Sources/Client/Client/API/Client+Channels.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Client+Channels.swift
+//  StreamChatCore
+//
+//  Created by Alexey Bukhtin on 09/01/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+// MARK: Channels Requests
+
+public extension Client {
+    
+    /// A message search.
+    /// - Parameters:
+    ///   - filter: a filter for channels, e.g. .key("members", .in(["john"]))
+    ///   - query: a search query.
+    ///   - pagination: a pagination. It works via the standard limit and offset parameters.
+    ///   - completion: a completion block with `[Message]`.
+    @discardableResult
+    func search(filter: Filter,
+                query: String,
+                pagination: Pagination = .channelsPageSize,
+                _ completion: @escaping Client.Completion<[Message]>) -> URLSessionTask {
+        if query.isBlank {
+            completion(.failure(.channelsSearchQueryEmpty))
+            return .empty
+        }
+        
+        if case .none = filter {
+            completion(.failure(.channelsSearchFilterEmpty))
+            return .empty
+        }
+        
+        let query = SearchQuery(filter: filter, query: query, pagination: pagination)
+        
+        return request(endpoint: .search(query)) { (result: Result<SearchResponse, ClientError>) in
+            completion(result.map({ $0.messages.compactMap({ $0["message"] }) }))
+        }
+    }
+    
+    /// Requests channels with a given query.
+    /// - Parameters:
+    ///   - query: a channels query (see `ChannelsQuery`).
+    ///   - completion: a completion block with `[ChannelResponse]`.
+    @discardableResult
+    func channels(query: ChannelsQuery, _ completion: @escaping Client.Completion<[ChannelResponse]>) -> URLSessionTask {
```

Comment:
on all the other clients we call this `queryChannels` or `query_channels`

Reply:
done